clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java

```diff
@@ -531,6 +531,10 @@ protected ConsumerConfig(Map<?, ?> props, boolean doLog) {
         return CONFIG.names();
     }
 
+    public static ConfigDef configDef() {
+        return  CONFIG;
```

Comment:
Sorry, I saw this late @rhauch and @mageshn. We should not do this as these types are mutable and anyone could change the shared definition.

Reply:
That's a good point @ijuma. What do you think about returning a copy  like 
`    public static ConfigDef configDef() {return  new ConfigDef(CONFIG);
    }`